src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Media_Imaging/Given_BitmapSource.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Markup;
+using Windows.UI.Xaml.Media.Imaging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Private.Infrastructure;
+
+namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml_Media_Imaging
+{
+	[TestClass]
+	public class Given_BitmapSource
+	{
+		[TestMethod]
+		[RunsOnUIThread]
+		public async Task When_SetSource_Then_StreamClonedSynchronously()
+		{
+			var sut = new BitmapImage();
+			var stream = new Given_BitmapSource_Stream();
+			var raStream = stream.AsRandomAccessStream();
+
+			var success = false;
+			try
+			{
+				sut.SetSource(raStream);
+			}
+			catch (Given_BitmapSource_Exception ex) when (ex.Caller is nameof(Given_BitmapSource_Stream.Read))
+			{
+				success = true;
+			}
+
+			Assert.IsTrue(success);
+		}
+
+		[TestMethod]
+		[RunsOnUIThread]
+		public async Task When_SetSourceAsync_Then_StreamClonedSynchronously()
+		{
+			var sut = new BitmapImage();
+			var stream = new Given_BitmapSource_Stream();
+			var raStream = stream.AsRandomAccessStream();
+
+			var success = false;
+			try
+			{
+				sut.SetSourceAsync(raStream);
+			}
+			catch (Given_BitmapSource_Exception ex) when (ex.Caller is nameof(Given_BitmapSource_Stream.Read))
+			{
+				success = true;
+			}
```

Comment:
Very strange pattern to rely on exceptions to validate a method is called.  There's features for that in mocking frameworks...

Reply:
Yes ... except it's a runtime test, there isn't any setup and I'm pretty sure that none exists for iOS which does not allow `Reflection.Emit` ;)